### PR TITLE
Added Modulestest readme and fixes

### PR DESCRIPTION
**Pull request by AhmedBM**
## Description

* Made `testplate-json` load from the root application path
* Added basic README.md for modulestest with usage examples and prerequisites
* Removed CMake warning for `jq` (needed to generate testplate.json)
* Added warning in `modulestest` when `testplate.json` is missing.

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] I added unit-tests to validate my changes. All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/modules/test/main.cpp [hunk 1]
```diff
@@ -25,8 +32,24 @@ TestRecipes LoadValuesFromConfiguration(std::stringstream& ss, std::string modul
 {
     TestRecipes testRecipes = std::make_shared<std::vector<TestRecipe>>();
     JSON_Value *root_value;
-    std::cout << "Using test recipes: " << g_defaultPath << std::endl;
-    root_value = json_parse_file_with_comments(g_defaultPath.c_str());
+    
+    char buf[PATH_MAX];
```
Comment (MariusNi) [inline src/modules/test/main.cpp:36]: >char buf[PATH_MAX]; [](http://example.com/codeflow?start=4&length=19)

= {0};

File: src/modules/test/main.cpp [hunk 2]
```diff
@@ -25,8 +32,24 @@ TestRecipes LoadValuesFromConfiguration(std::stringstream& ss, std::string modul
...
+    if (res == nullptr)
+    {
+        TestLogError("Could not find the executable path");
+        return testRecipes;
+    }
+    std::string fullPath(buf);
+    fullPath = fullPath.substr(0, fullPath.find_last_of("/") + 1) + g_defaultPath;
+    if (!std::filesystem::exists(fullPath.c_str()))
+    {
+        TestLogError("Could not find test configuration: %s\n May be missing 'jq' application, install application and perform CMake rebuild (configure+build).", fullPath.c_str());
```
Comment (MariusNi) [inline src/modules/test/main.cpp:47]: >May be missing 'jq' application [](http://example.com/codeflow?start=62&length=31)

Suggest:

"jq may be missing,"

Comment (MariusNi) [inline src/modules/test/main.cpp:47]: >perform CMake rebuild (configure+build) [](http://example.com/codeflow?start=119&length=39)

Just say "and rebuild"

File: src/modules/test/main.cpp [hunk 3]
```diff
@@ -211,11 +232,21 @@ int main(int argc, char **argv)
...
-    std::cout << "Test Summary: " << std::endl << ss.str();
-    std::cout << "[==========]"   << std::endl;
+        std::cout << "[==========]" << std::endl;
+        std::cout << "Test Summary: " << std::endl
+                  << ss.str();
+        std::cout << "[==========]" << std::endl;
+    }
+    else
+    {
+        std::cerr << "No tests found!" << std::endl;
```
Comment (MariusNi) [inline src/modules/test/main.cpp:247]: >! [](http://example.com/codeflow?start=36&length=1)

No need for !

File: src/modules/test/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Modulestest
+The modulestest application is OSConfig’s module testing framework, it allows module authors to write test recipes which in turn, test the modules functionality and conformity (against the published MIM) of the Management Module.
```
Comment (MariusNi) [inline src/modules/test/README.md:2]: >test [](http://example.com/codeflow?start=130&length=4)

"drive testing of the module functionality"

File: src/modules/test/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# modulestest
+The modulestest application is OSConfig’s module testing framework, it allows module authors to write test recipes which in turn, drive testing of the module functionality and conformity (against the published MIM) of the Management Module.
```
Comment (robertschaedler3) [inline src/modules/test/README.md:2]: nit: add \`...\` around `modulestest` everywhere

**Review by MariusNi — DISMISSED** [review]
:shipit:

**Review by robertschaedler3 — APPROVED** [review]

**Pull request merged**